elastalert/elastalert.py

```diff
@@ -426,11 +423,10 @@ def run_rule(self, rule, endtime, starttime=None):
             if not self.run_query(rule, rule['starttime'], tmp_endtime):
                 return 0
             rule['starttime'] = tmp_endtime
+            rule['type'].garbage_collect(endtime)
         if not self.run_query(rule, rule['starttime'], endtime):
             return 0
 
-        rule['type'].garbage_collect(endtime)
```

Comment:
This shouldn't be removed. If the loop is never entered, run_query will still be called and garbage_collect still needs to be run.